docs/contributing.rst

```diff
@@ -161,29 +161,61 @@ Any extra texts (print statements and so on) should be removed.
 Tests coverage
 --------------
 
-We are trying hard to have good test coverage; please don't make it worse.
+We use *codecov.io* as an indispensable tool for analysing our coverage
+results. Visit https://codecov.io/gh/aio-libs/aiohttp to see coverage
+reports for the master branch, history, pull requests etc.
 
-Use:
+We'll use an example from a real PR to demonstrate how we use this.
+Once the tests run in a PR, you'll see a comment posted by *codecov*.
+The most important thing to check here is whether there are any new
+missed or partial lines in the report:
 
-.. code-block:: shell
+.. image:: contributing-cov-comment.png
 
-   $ make cov-dev
+Here, the PR has introduced 1 miss and 2 partials. Now we
+click the link in the comment header to open the full report:
 
-to run test suite and collect coverage information. Once the command
-has finished check your coverage at the file that appears in the last
-line of the output:
-``open file:///.../aiohttp/htmlcov/index.html``
+.. image:: contributing-cov-header.png
+   :alt: Codecov report
+
+Now, if we look through the diff under 'Files changed' we find one of
+our partials:
+
+.. image:: contributing-cov-partial.png
+   :alt: A while loop with partial coverage.
+
+In this case, the while loop is never skipped in our tests. This is
+probably not worth writing a test for (and may be a situation that is
+impossible to trigger anyway), so we leave this alone.
```

Comment:
One way of dealing with such cases is using `# pragma: no cover`. This would allow setting higher required coverage level eventually, like this: https://adamj.eu/tech/2019/04/30/getting-a-django-application-to-100-percent-coverage/#the-report-section.

Reply:
Yep, I'm aware of this, but have only been using such comments on things which are definitely never going to be covered in a test (like `if TYPE_CHECKING`). In this example, I'm not 100% sure that the condition can never happen, it just didn't seem worth even trying to write a test for it.

Reply:
I'll leave this comment for now, as we don't really follow this practice currently. Can return to it if we decide to start enforcing it.